Code/Framework/AzCore/Tests/Math/LineSegmentTests.cpp

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/LineSegment.h>
+#include <AzCore/Math/Ray.h>
+#include <AzCore/Math/Vector3.h>
+#include <AzCore/UnitTest/TestTypes.h>
+
+namespace UnitTest
+{
+
+    TEST(MATH_LineSegment, LineSegment_CreateFromStartAndEnd)
+    {
+        AZ::LineSegment segment(AZ::Vector3(0.0f, 15.5f, 12.0f), AZ::Vector3(12.0f, 0.0f, 4.0f));
+        EXPECT_TRUE(segment.GetStart().IsClose(AZ::Vector3(0.0f, 15.5f, 12.0f)));
+        EXPECT_TRUE(segment.GetEnd().IsClose(AZ::Vector3(12.0f, 0.0f, 4.0f)));
```

Comment:
update: I'd definitely recommend updating all these to use `EXPECT_THAT(..., IsClose(...))` as if there's a failure you'll then get something much more useful printed to the console (insteado f just 'false').

Reply:
Great to see the tests being added though! Awesome work 🙂